tests/test_shared.py

```diff
@@ -26,3 +32,43 @@ def test_factorize_assortative_variables(initial_states, assort_by, expected):
     )
 
     assert set(group_code_values) == set(expected)
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "x, seed, expected", [(0, None, False), (1, None, True), (0.7, 0, True)]
+)
+def test_boolean_choice(x, seed, expected):
+    np.random.seed(seed)
+    result = boolean_choice(x)
+    assert result is expected
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "choices, probs, seed, expectation, expected",
+    [
+        (3, [0, 0, 1], None, does_not_raise(), 2),
+        ([0, 1, 2], [0, 1, 0], None, does_not_raise(), 1),
+        (np.arange(3), [1, 0, 0], None, does_not_raise(), 0),
+        (3, None, 0, does_not_raise(), 1),
```

Comment:
Why is 1 the expected result here?

Reply:
I set the seed to zero which results in 1. It is not a deterministic edge case if this was your question.